src/DataFixtures/SlotAvailabilityGuesser.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\DataFixtures;
+
+class SlotAvailabilityGuesser
+{
+    const WEEKEND_CHANCES_MODIFIER = 15;
+    const RATE_OF_AVAILABLE_PERCENTAGE = 5;
+
+    const SLOT_AVAILABLE_DAY = [
+        '6', // Saturday
+        '7',  // Sunday
+    ];
+
+    private int $availableSlotCount = 0;
+    private int $availableSlotPercentageWithPrevious = 100;
+    private ?\DateTimeImmutable $lastSlot = null;
+
+    public function guessAvailableSlot(\DateTimeImmutable $slot): bool
+    {
+        $rand = rand(1, 100);
+        $chance = $this->getSlotChances($slot);
+
+        if ($this->availableSlotCount > 0) {
+            $chance = $this->availableSlotPercentageWithPrevious;
+
+            // Decreasing chances to be available
+            $this->availableSlotPercentageWithPrevious -= self::RATE_OF_AVAILABLE_PERCENTAGE;
+        }
+
+        if ($rand <= $chance) {
+            ++$this->availableSlotCount;
+            $this->lastSlot = $slot;
+
+            return true;
+        }
+
+        $this->resetGuesser();
+
+        return false;
+    }
+
+    /**
+     * @param int $resetLastSlot
+     */
+    public function resetGuesser($resetLastSlot = 1): void
+    {
+        if ($resetLastSlot) {
+            $this->lastSlot = null;
+        }
+        $this->availableSlotCount = 0;
+        $this->availableSlotPercentageWithPrevious = 100;
+    }
+
+    private function getSlotChances(\DateTimeImmutable $slot): int
+    {
+        $time = (int) $slot->format('H');
+
+        // Avoid 2 booking same day
+        if ($this->lastSlot && ((int) $slot->format('d') <= (int) $this->lastSlot->format('d'))) {
+            return -1;
+        }
+
+        if ($time >= 0 && $time <= 8) {
```

Comment:
As suggestion, using switch syntax instead of multiple if/elseif would decrease the complexity:
```php
switch (true) {
    case 0 <= $time && 8 >= $time:
        //...
}
```

Reply:
Ok